File: Java-IDEA-Projects-main/Java-IDEA-Projects-main/Algorithms/Sorting/src/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {
    public static void main(String[] args) {
        int[] arr = {4, 1, 3, 2, 5};
        bubbleSort(arr);
        System.out.println(Arrays.toString(arr));
    }

    private static void bubbleSort(int[] arr) {
        boolean swapped;
        for (int i = 0; i < arr.length; i++) {
            swapped = false;
            for (int j = 0; j < arr.length - i - 1; j++) {
                // swap logic, if the previous element is greater than the next element we swap the elements.
                    if(arr[j] > arr[j + 1]){
                        int temp = arr[j];
                        arr[j] = arr[j + 1];
                        arr[j + 1] = temp;
                        swapped = true;
                    }
            }

            if(!swapped){
                break;
            }
        }

    }

    private static void swap(int i, int i1) {
        int temp = i;
        i = i1;
        i1 = temp;
    }


}

File: CPrograms-master/CPrograms-master/bubble sort.c
#include<stdio.h>

void bubble_sort(int [],int);

int main()
{
int A[]={34,15,29,8},i;
bubble_sort(A,4);
for(i=0;i<=3;i++)
{
printf("%d ",A[i]);
}
return 0;
}


void bubble_sort(int A[], int N)
{
int i,round,temp;
for(round=0;round<=N-1;round++)
{
for(i=0;i<=N-1-round;i++)
{
if(A[i]>A[i+1])
{
temp=A[i];
A[i]=A[i+1];
A[i+1]=temp;
}
}
}
}
